### PR TITLE
[ML] Re-enable sdk logging

### PR DESCRIPTION
**Pull request by diondrapeck**
# Description

Related PR: https://github.com/Azure/azure-sdk-for-python/pull/26583/files

SDK logging needed to be switched from applicationinsights to opencensus-ext-azure since the former is deprecated and all dependencies needed to be GA-ed once v2 was GA-ed. This PR reenables telemetry logging in the SDK for Jupyter Notebooks only (other logging in Azure SDKs is not allowed).

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/_telemetry/logging_handler.py [hunk 1]
```diff
@@ -51,308 +44,99 @@
...
+def get_appinsights_log_handler(
+     user_agent,
+     *args, # pylint: disable=unused-argument
+     instrumentation_key=None,
+     component_name=None,
+     **kwargs
+ ):
+    """Enable the OpenCensus logging handler for specified logger and instrumentation key to send info to AppInsights.
+
+    Enable diagnostics collection with the :func:`azureml.telemetry.set_diagnostics_collection` function.
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_telemetry/logging_handler.py:94]: Is this legacy comment from v1 SDK?

File: sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_logger_utils.py [hunk 2]
```diff
@@ -20,11 +20,11 @@ def initialize_logger_info(module_logger: logging.Logger, terminator="\n") -> No
 
 class OpsLogger:
     def __init__(self, name: str):
-        # self.logger: logging.Logger = logging.getLogger(AML_INTERNAL_LOGGER_NAMESPACE + name)
-        # self.logger.propagate = False
+        self.logger: logging.Logger = logging.getLogger(AML_INTERNAL_LOGGER_NAMESPACE + name)
```
Comment (luigiw) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_logger_utils.py:23]: Minor: the variable name `logger` is too generic and can cause confusion. Consider a more specific name.

**Comment by luigiw** [comment]
I think it was also asked in our team meeting today. Should we have a way for customers to opt out telemetry collection?

**Comment by diondrapeck** [comment]
> I think it was also asked in our team meeting today. Should we have a way for customers to opt out telemetry collection?

They can opt out by setting the environment variable 'AZUREML_SDKV2_TELEMETRY_OPTOUT' to '1' or 'true'

File: sdk/ml/azure-ai-ml/azure/ai/ml/_telemetry/logging_handler.py (hunk 1 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_telemetry/logging_handler.py:94]: Yes, good catch. I removed it and updated the docstring.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_logger_utils.py (hunk 2 above)
Comment (diondrapeck) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_utils/_logger_utils.py:23]: Fair. I renamed to "package_logger" to differentiate from "module_logger"

**Comment by luigiw** [comment]
> > I think it was also asked in our team meeting today. Should we have a way for customers to opt out telemetry collection?
> 
> They can opt out by setting the environment variable 'AZUREML_SDKV2_TELEMETRY_OPTOUT' to '1' or 'true'

This is cool. We may need to consider a more explicit flag. E.g. a variable to MLClient. The reason being env vars may have constrains that are out of control of a user sometime.

**Comment by luigiw** [comment]
Another thing I just remembered is that we need to add a disclaimer on telemetries to README.md. Sophia had it when she added telemetries. but it was removed later by accident. Here's original disclaimer Sophia had. https://pypi.org/project/azure-ai-ml/0.1.0b1/

**Review by luigiw — APPROVED** [review]

**Pull request merged**